python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -990,6 +990,13 @@ def __sub__(self, other: "PartitionsSubset") -> "PartitionsSubset[T_str]":
             set(self.get_partition_keys()).difference(set(other.get_partition_keys()))
         )
 
+    def __and__(self, other: "PartitionsSubset") -> "PartitionsSubset[T_str]":
+        if self is other:
+            return self
+        return self.partitions_def.empty_subset().with_partition_keys(
```

Comment:
Any perf impact you'd worry about here?

Reply:
I think calling this function is fine in the context of the asset backfill page (since its basically the same logic as before).

But if this operation (or the `__or__` and `__sub__` operations above) were called repeatedly on the same subset, it would be worse than fetching the set of partitions once and doing set operations from there